In [27]:
pip install pandas numpy

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.


In [33]:
import pandas as pd


In [29]:
df = pd.read_csv('vendas_ficticias.csv')
df


Data da venda  Valor da venda Vendedor  Quantidade vendida    Produto
0     2024-01-29          212.80    Pedro                   8  Produto D
1     2024-05-19          428.04     João                  16  Produto E
2     2024-02-19          982.79      Ana                  20  Produto A
3     2024-02-15          127.86    Pedro                   1  Produto B
4     2024-01-13          429.88    Maria                   6  Produto E
..           ...             ...      ...                 ...        ...
95    2024-03-13           78.17     João                  18  Produto A
96    2024-04-19          865.72    Maria                  14  Produto D
97    2024-01-19          117.36   Carlos                  12  Produto E
98    2024-01-13          934.73    Pedro                   4  Produto E
99    2024-03-26          349.69      Ana                  11  Produto E

[100 rows x 5 columns]

In [8]:
import pandas as pd

# Carregar o DataFrame
df = pd.read_csv('vendas_ficticias.csv')

# Expandir a exibição de linhas e colunas
pd.set_option('display.max_rows', 100)  # Defina o número máximo de linhas a serem exibidas
pd.set_option('display.max_columns', 100)  # Defina o número máximo de colunas a serem exibidas

# Exibir o DataFrame
print(df)

   Data da venda  Valor da venda Vendedor  Quantidade vendida    Produto
0     2024-01-29          212.80    Pedro                   8  Produto D
1     2024-05-19          428.04     João                  16  Produto E
2     2024-02-19          982.79      Ana                  20  Produto A
3     2024-02-15          127.86    Pedro                   1  Produto B
4     2024-01-13          429.88    Maria                   6  Produto E
5     2024-03-02          259.37    Maria                  16  Produto E
6     2024-04-28          389.57     João                  13  Produto D
7     2024-04-19          866.23      Ana                   1  Produto D
8     2024-01-20          854.59    Maria                  19  Produto C
9     2024-01-03          267.65    Pedro                  14  Produto C
10    2024-05-31           66.14    Pedro                   7  Produto A
11    2024-03-15          463.15      Ana                   6  Produto E
12    2024-03-17          170.13    Pedro          

In [34]:
# cálculo da média usando a função mean
media_vendas = df['Valor da venda'].mean()
print(media_vendas)


553.2947


In [16]:
# cálculo usando a função median
mediana_vendas = df['Valor da venda'].median()
print(mediana_vendas)

569.535


In [17]:
min_vendas = df['Valor da venda'].min()
print(min_vendas)

55.02


In [19]:
max_vendas = df['Valor da venda'].max()
print(max_vendas)

987.42


In [20]:
desvio_padrao_vendas = df['Valor da venda'].std()
print(desvio_padrao_vendas)

281.06073685041633


In [22]:
quantidade_vendida_media = df['Quantidade vendida'].mean()
print(quantidade_vendida_media)

9.9


In [23]:
quantidade_vendida_mediana = df['Quantidade vendida'].median()
print(quantidade_vendida_mediana)

10.0


In [24]:
quantidade_vendida_min = df['Quantidade vendida'].min()
print(quantidade_vendida_min)

1


In [25]:
quantidade_vendida_max = df['Quantidade vendida'].max()
print(quantidade_vendida_max)

20


In [26]:
quantidade_vendida_desvio_padrao = df['Quantidade vendida'].std()
print(quantidade_vendida_desvio_padrao)

5.880115438982264


In [37]:
# cálculo usando a função mode
moda_vendas = df['Produto'].mode()
print(moda_vendas)

0    Produto D
Name: Produto, dtype: object


Resultado sobre as vendas:
A média e a mediana são próximas, o que sugere que há uma boa distribuição dos valores das vendas (media: 553.29 e mediana 569.54 )
o desvio padrão mostrou-se consideravelmente alto (281.060737), o que mostra uma variação nos valores das vendas, que pode ser observado também ao analisar o valor mínimo 55.02 e o valor máximo de 987.42.

Resultado sobre a quantidade
A média e a mediana são bem próximas, (9.9 e 10), mantendo uma constância nas quantidades. 
A quantidade de produtos vendidos sofreu uma variação entre a quantidade mínima de 1 e a máxima de 20 produtos. O desvio padrão nesse caso foi de 5.880115.

In [38]:
# função describe Pandas
df.describe().T

count      mean         std    min       25%      50%  \
Valor da venda      100.0  553.2947  281.060737  55.02  335.4675  569.535   
Quantidade vendida  100.0    9.9000    5.880115   1.00    4.0000   10.000   

                         75%     max  
Valor da venda      813.9275  987.42  
Quantidade vendida   15.2500   20.00